docs/guide/exe_verilator.rst

```diff
@@ -344,6 +344,15 @@ Summary:
 
        touch foo.v ; verilator -E --dump-defines foo.v
 
+.. option:: --dump-dfg
+
+   Rarely needed.  Enable writing DFG optimizer .dot files with dumping level 1,
+
+.. option:: --dump-dfgi <level>
+
+   Rarely needed - for developer use.  Set internal DFG dumping level
+   globally to a specific dumping level.
+
```

Comment:
I am not sure this is an improvement, as I would still have to document the special use of --debugi-<Source>, but more importantly, if you turn this up even moderately high, you will get *a lot* of dot files, which are expensive to output and are almost always noise (unless specifically debugging DFG optimizations), so I don't think you want them if you just do --debug (which sets default level to 3), which your suggested change would do?

Reply:
I was assuming you would then only dump the files if the debug level was 9.  All across the program if you use debug level 9 you'll may get thousands of dot files.

Reply:
Perhaps "--dump-treei-V3Dfg 9" would suit us both?  That gives you control, and resolves my worry that we start propagating many debug flags in some non-consistent/extendable way?

Reply:
(That is there's a filename extension implemented like --debugi- supports.)

Reply:
I appreciate what you are trying to do here. My problem with that proposal is that there are no trees involved here, so --dump-treeei would be a misnomer. Inaccurate naming is one of the things that I think makes code and tools a lot harder to understand and use. If it's a cat then let's not call it a dog and then make people read 200 lines of code to figure out that the reason we feed the dog cat food is because it really is a cat.

As a sidenote: the dump and debug options are already pretty inconsistent (you need various debug levels enabled as the dumps are under `if (debug() >= n)` etc.). The dump-dfg stuff I added in this patch is at least orthogonal to the other switches in that respect.

Btw I'm all for clarity and consistency, but reusing a switch that refers to one internal representation to be used by a different IR is not helping either.

Reply:
I see your points... Do you have suggestion that won't require we add new flags for the next new fancy optimization with unique dumpers?

Reply:
We could go along the lines you were suggesting, just drop the 'tree' bit? So we have `-dumpi-<thing>` where \<thing\> can be either:
- file basenam like V3Foo, or
- magic word like tree/graph/dfg

Reply:
Works for me!

Reply:
See #3630